### PR TITLE
Decimal128 and Decimal256 support

### PR DESCRIPTION
**Pull request by changhiskhan**
addresses #577

- [x] type conversion
- [x] read
- [x] write

### CODE REVIEW

File: rust/src/datatypes.rs [hunk 1]
```diff
@@ -76,6 +76,7 @@ impl TryFrom<&DataType> for LogicalType {
             DataType::Float16 => "halffloat".to_string(),
             DataType::Float32 => "float".to_string(),
             DataType::Float64 => "double".to_string(),
+            DataType::Decimal128(precision, scale) => format!("decimal128:{precision}:{scale}"),
```
Comment (eddyxu) [inline rust/src/datatypes.rs:79]: Arrow has a Decimal256 as well, might wanna to add it?

File: rust/src/io/writer.rs [hunk 2]
```diff
@@ -290,6 +291,7 @@ mod tests {
             ArrowField::new("i", DataType::Int64, true),
             ArrowField::new("f", DataType::Float32, false),
             ArrowField::new("b", DataType::Utf8, true),
+            ArrowField::new("d", DataType::Decimal128(7, 3), false),
```
Comment (eddyxu) [inline rust/src/io/writer.rs:294]: do we need to set byte width for this type, so that plain encoder can pick it up?

Comment (changhiskhan) [reply rust/src/io/writer.rs]: it was already there:

Comment (changhiskhan) [reply rust/src/io/writer.rs]:
```

    fn byte_width(&self) -> usize {
        match self {
            Self::Int8 => 1,
            Self::Int16 => 2,
            Self::Int32 => 4,
            Self::Int64 => 8,
            Self::UInt8 => 1,
            Self::UInt16 => 2,
            Self::UInt32 => 4,
            Self::UInt64 => 8,
            Self::Float16 => 2,
            Self::Float32 => 4,
            Self::Float64 => 8,
            Self::Date32 => 4,
            Self::Date64 => 8,
            Self::Time32(_) => 4,
            Self::Time64(_) => 8,
            Self::Decimal128(_, _) => 16,
            Self::Decimal256(_, _) => 32,
            Self::FixedSizeBinary(s) => *s as usize,
            Self::FixedSizeList(dt, s) => *s as usize * dt.data_type().byte_width(),
            _ => panic!("Does not support get byte width on type {self}"),
        }
    }
```

Comment (changhiskhan) [reply rust/src/io/writer.rs]: in arrow.rs

File: rust/src/datatypes.rs (hunk 1 above)
Comment (changhiskhan) [reply rust/src/datatypes.rs]: added

**Review by eddyxu — APPROVED** [review]
🚢

**Pull request merged**